src/system/sys_profiler.hh

```diff
@@ -294,7 +357,19 @@ public:
      * Start the timer. If it is already started, just increase number of starts (recursions) and calls.
      */
     void start();
-
+    /**
+     * Pause current timer, save measured petsc memory information util resume
```

Comment:
Describe better how this mechanism interact with PETSC calls. Possibly pause and resume may be protected (since Profiler is a friend class) and the user should not call them directly.